### PR TITLE
Shorten --max-time in tests and use a more stable error check

### PR DESCRIPTION
**Pull request by alex-kattathra-johnson**
- fixes flakey tests from solving #14241

# Description

@sholderbach: This is a preliminary fix for the flaky tests and also shortened the `--max-time` in the tests.

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

**Comment by sholderbach** [comment]
As the observed test failure was transient I am curious if the changed assertion will catch the rarer case. Example: https://github.com/nushell/nushell/actions/runs/12167920980/job/33937573057

**Comment by alex-kattathra-johnson** [comment]
I think so since the error message remains `timed out reading response` regardless of the transient changes in the diagnostic code.

**Comment by sholderbach** [comment]
Is it catching the same failure mode (i.e. truly timing out) or is that failure pointing that the connection has been closed by `mockito` because there is a delay for `nu` starting up? So then we could make the ratio of timeouts greater?

**Comment by alex-kattathra-johnson** [comment]
@sholderbach: It looked like we were throwing ioerr for everything once it had become a `Spanned`. The transient nature of the failures might have been because the short timeout was causing the error to be propagated much higher up in the chain where it was being detected as a `NetworkFailure`.

File: crates/nu-protocol/src/errors/shell_error.rs [hunk 1]
```diff
@@ -1557,12 +1557,15 @@ impl From<Spanned<io::Error>> for ShellError {
                     msg: "unknown error".into(),
                     span,
                 },
-            }
-        } else {
-            Self::IOErrorSpanned {
+            },
+            io::ErrorKind::TimedOut => Self::NetworkFailure {
```
Comment (sholderbach) [inline crates/nu-protocol/src/errors/shell_error.rs:1561]: Wondering if we ever might run into a non-networking timeout on some obscure file system?

File: crates/nu-command/tests/commands/network/http/put.rs [hunk 2]
```diff
@@ -171,18 +171,25 @@ fn http_put_timeout() {
     let _mock = server
         .mock("PUT", "/")
         .with_chunked_body(|w| {
-            thread::sleep(Duration::from_secs(1));
+            thread::sleep(Duration::from_secs(10));
```
Comment (sholderbach) [inline crates/nu-command/tests/commands/network/http/put.rs:174]: Wondering if the `server` joins on those threads on drop or if they get ditched. But for now more reliable CI takes priority over trying to squeeze out the fastest test run (looking at the plugin timeout tests we have bigger fish to fry in that regard)

**Review by sholderbach — COMMENTED** [review]
Thanks! Let's give this a try

**Pull request merged**